GVFS/GVFS.Common/GVFSPlatform.cs

```diff
@@ -73,6 +73,10 @@ public static void Register(GVFSPlatform platform)
             ITracer tracer,
             string lockPath);
 
+        public abstract ProductUpgraderPlatformStrategy CreateProductUpgraderPlatformInteractions(
```

Comment:
I think the `ProductUpgraderPlatformStrategy` name is overly verbose and at the same time not clear. I don't think the `Interactions` suffix is any better... Still thinking for a better name.